plutus-use-cases/test/Spec/TokenAccount.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Spec.TokenAccount(tests) where
+
+import           Test.Tasty
+
+import           Language.Plutus.Contract
+import           Language.Plutus.Contract.Test
+import           Language.PlutusTx.Lattice
+import qualified Ledger.Ada                                            as Ada
+import           Ledger.Value                                          (Value)
+import qualified Ledger.Value                                          as Value
+
+import qualified Language.PlutusTx.Coordination.Contracts.TokenAccount as Accounts
+
+tests :: TestTree
+tests = testGroup "token account"
+    [ checkPredicate "Create a token account"
+        initialiseAccount
+        (assertNoFailedTransactions
+        /\ assertDone w1 (const True) "wallet 1 should be done"
+        /\ walletFundsChange w1 (Ada.lovelaceValueOf (-10) <> theToken))
+        (handleBlockchainEvents w1)
+
+    , checkPredicate "Redeem all funds"
+        (initialiseAccount >>= Accounts.redeem (walletPubKey w1))
+        (assertNoFailedTransactions
+        /\ assertDone w1 (const True) "wallet 1 should be done"
+        /\ walletFundsChange w1 theToken)
```

Comment:
I was a bit confused by no ada check here, but then I realised you're claiming the money back to the original owner. Maybe have a test where you send the token to someone else and they redeem?

Reply:
Done